collector/collector.cpp

```diff
@@ -251,13 +228,15 @@ int main(int argc, char** argv) {
 
   CollectorService collector(config, &g_control, &g_signum);
 
-  if (!collector.InitKernel()) {
-    g_startup_diagnostics.Log();
+  if (!collector.InitKernel(args->GRPCServer(), candidates)) {
```

Comment:
this is maaaaybe a nit, but I find it very odd at this level that we're passing the address of the GRPC server to a function that is meant to initialize kernel components - from an encapsulation perspective it feels a bit off. I'm not opposed to the collector service doing the download, but can it get this information in a different way that doesn't require the caller to know about it? Specifically, can we store the GRPC address in the `CollectorConfig` which is accessible from the `CollectorService`?

Reply:
Yeah I don't like this approach either. My initial thought was to have an independent function do the download but I turned out to be a bit awkward too. I'll take another look, maybe we just want some sort of `MakeCollectorService` function that takes whatever we need (I also want to cut down on the amount of stuff that is needed to be copied).

Reply:
Ended up adding a `SetupKernelDriver` that takes in a `CollectorService` object alongside the grpc server address and the configuration, and in turn handles everything from finding/downloading the drivers to calling `collector.InitKernel`, hopefully this is a bit better than the previous approach.

Reply:
Yeah that approach looks good!